src/sql/workbench/contrib/query/test/browser/queryInputFactory.test.ts

```diff
@@ -141,6 +141,217 @@ suite('Query Input Factory', () => {
 		assert(connectionManagementService.numberConnects === 0, 'Convert input should not have been called connect when no global connections exist');
 	});
 
+	test('sync query editor input can only be connected once when global connection exists', () => {
```

Comment:
What about tests when a global connection doesn't exist?

Reply:
Those tests already exist in this file.